students/perrymant/blog-01-hello-world-deconstructed.md

```diff
@@ -0,0 +1,347 @@
+Besides the `public` modifier, there is also `private`, `protected` and the `package-private` modifier.
+This last one is the default modifier, that is to say that when you don't add any of the other modifier keywords, then the class is automatically designated as `package-private`, which is what the example `Frog` class above is.
+These access modifiers apply not only to classes, but also to methods, constructors and data members.
+
+![Modifiers](assets/blog-01-hello-world-deconstructed/accessmod.png)
+[credit](https://stackoverflow.com/a/215505)
+
+#### 1. `public` access modifier
+
+The access modifier `public` is fairly self-descriptive in that it makes the class publicly availiable to other classes.
+
+An important thing to note is that a file may only have one `public class`.
+So it's OK to have other classes in the file, just as long as they have a different access modifier, which means I can put in a `package private` (see below #3) class called `FrogFeed` in the same file as the `Frog`:
+```java
++---Frog.java---------------+
+|                           |
+|  public class Frog {      |
+|  }                        |
+|                           |
+|  class FrogFeed { |
+|  }                        |
++---------------------------+
+```
+#### 2. `protected` access modifier
+
+The `protected` access modifier can't actually be applied to classes, but it can be used with methods, constructors and data members, so we'll look at it here, and start by understanding how we organise classes in packages and files.
+
+The way we organise our classes is through files, packages and folders.
+We've already seen how .java files contain the classes, and you're probably familiar with folders (or directories) because your computer's Operating System uses these to create a hierarchy for storing files.
+Packages are exactly the same as folders, with the only difference being that they are the folders within the `src` folder:
+```java
+Documents
+└── pizzeria                                <- folder
+    └── src                                 <- folder
+        └── com                             <- package
+            └── example                     <- package
+                ├── appliances              <- package
+                │   ├── Oven.java
+                │   └── Refrigerator.java
+                ├── food                    <- package
+                │   ├── Antipasti.java
+                │   ├── Pasta.java
+                │   ├── Pizza.java
+                │   └── ingredients         <- package
+                │       ├── Cheese.java
+                │       ├── Flour.java
+                │       └── Tomato.java
+                └── restaurant              <- package
+                    ├── Kitchen.java
+                    └── Toilet.java
+```
+Packages are used for organising related classes, and to put a class in a package, you need to add the line
+
+`package `*packagename*`;`
```

Comment:
I'm guessing this doesn't render the way you intended?
Might be best to use code blocks with Java syntax here.